contacts/views.py

```diff
@@ -1,12 +1,16 @@
+from django.contrib.auth.decorators import login_required
 from django.http import HttpResponseRedirect
 from django.shortcuts import render
 from django.urls import reverse
 from django.views import generic, View
+from django.utils.decorators import method_decorator
 
 from contacts.forms import NewContactForm, NewAddressForm
 from .models import Contact, Address
 
 
+
+@method_decorator(login_required, name='dispatch')
```

Comment:
we don't need to do it when we're adding the decorator around url.
these both method are alternatives of each other!